packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -271,107 +136,138 @@ export class SwatchColorPickerBase extends React.Component<ISwatchColorPickerPro
-    const { onCellFocused } = this.props;
+  // Callback passed to the GridCell class that will trigger the onCellFocus callback of the SwatchColorPicker
+  const onGridCellFocused = (item?: IColorCellProps): void => {
+    const { onCellFocused } = props;
     if (onCellFocused) {
       if (item) {
-        this._cellFocused = true;
+        cellFocused = true;
         return onCellFocused(item.id, item.color);
       } else {
-        this._cellFocused = false;
+        cellFocused = false;
         return onCellFocused();
       }
     }
   };
 
-  /**
-   * Handle the click on a cell
-   * @param item - The cell that the click was fired against
-   */
-  private _onCellClick = (item: IColorCellProps): void => {
-    if (this.props.disabled) {
+  // Handle the click on a cell
+  const onCellClick = (item: IColorCellProps): void => {
+    if (props.disabled) {
       return;
     }
 
     const index = item.index as number;
 
-    // If we have a valid index and it is not already
-    // selected, select it
-    if (index >= 0 && index !== this.state.selectedIndex) {
-      if (this.props.onCellFocused && this._cellFocused) {
-        this._cellFocused = false;
-        this.props.onCellFocused();
+    // If we have a valid index and it is not already selected, select it.
+    if (index >= 0 && index !== selectedIndex) {
+      if (props.onCellFocused && cellFocused) {
+        cellFocused = false;
+        props.onCellFocused();
       }
 
-      if (this.props.onColorChanged) {
-        this.props.onColorChanged(item.id, item.color);
+      if (props.onColorChanged) {
+        props.onColorChanged(item.id, item.color);
       }
 
       // Update internal state only if the component is uncontrolled
-      if (this.props.isControlled !== true) {
-        this.setState({
-          selectedIndex: index,
-        });
+      if (props.isControlled !== true) {
+        setSelectedIndex(index);
       }
     }
   };
-}
+
+  // Render a color cell
+  const renderOption = (item: IColorCellProps): JSX.Element => {
+    return (
+      <ColorPickerGridCell
+        item={item}
+        idPrefix={id}
+        color={item.color}
+        styles={props.getColorGridCellStyles}
+        disabled={disabled}
+        onClick={onCellClick}
+        onHover={onGridCellHovered}
+        onFocus={onGridCellFocused}
+        selected={selectedIndex !== undefined && selectedIndex === item.index}
+        circle={cellShape === 'circle'}
+        label={item.label}
+        onMouseEnter={onMouseEnter}
+        onMouseMove={onMouseMove}
+        onMouseLeave={onMouseLeave}
+        height={props.cellHeight}
+        width={props.cellWidth}
+        borderWidth={props.cellBorderWidth}
+      />
+    );
+  };
+
+  if (colorCells.length < 1 || columnCount < 1) {
+    return null;
+  }
+
+  if (props.selectedId) {
+    currentSelectedIndex = getSelectedIndex(props.colorCells, props.selectedId);
+    setSelectedIndex(currentSelectedIndex);
```

Comment:
Pretty sure you shouldn't be setting state while rendering. (In this case it might even cause an infinite rendering loop if `props.selectedId` is defined!)

See overall review comment for what needs to be done instead.